### ISSUE TITLE
QST: TypeError: Object of type QVariant is not JSON serializable

### ISSUE THREAD

**Issue by vinirafaelsch** [geopandas/geopandas#2269]
I'm using Geopandas' `from_features()` method to transform an object of type QgsFeature from QGis into a new GeoDataFrame and it's working as expected.

However, when I try to use GeoDataFrame's `to_json()` method I'm getting the following error:

**TypeError: Object of type QVariant is not JSON serializable**

Note:

1 - I've already checked the variable's type before using `to_json()`, it's of type <class 'geopandas.geodataframe.GeoDataFrame'>

2 - How I'm loading QGis' QgsFeature object into Geopandas:
```
layer = self.iface.activeLayer()
selected_features = layer.selectedFeatures()
input = gpd.GeoDataFrame.from_features(selected_features, crs = iface.activeLayer().sourceCrs().authid())
```

3- I didn't do any manipulation with GeoDataFrame before using the `to_json()` method

4 - When I use `to_json()` in GeoSeries containing the geometries of this GeoDataFrame it works correctly. But I need all data stored in GeoDataFrame to be transformed into json.

Thanks

**Comment by martinfleis** [comment]
Hi, I have never tried that but it seems that within your GeoDataFrame is an object of type `QVariant`.  You need to convert it to something geopandas understands or drop the affected rows/columns. 

We can try to debug further but we would need a fully reproducible example (that snippet above is not).

**Comment by vinirafaelsch** [comment]
Hi @martinfleis , sorry for the delay in responding.

I was able to solve the problem based on what you told me. Basically, some of the columns came null from QGis and those columns assumed the `QVariant` type. So, in the case of my application these columns were not relevant to me, what I did was go through all the rows and all columns checking if the data was of type `QVariant`, if true, I chose to remove these columns.

I will close the question here, and thank you very much for the help. :)
